src/tests/class_tests/openms/source/IdentificationDataConverter_test.cpp

```diff
@@ -0,0 +1,166 @@
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Hendrik Weisser $
+// $Authors: Hendrik Weisser $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+#include <OpenMS/SYSTEM/SysInfo.h>
+#include <OpenMS/test_config.h>
+
+///////////////////////////
+
+#include <OpenMS/METADATA/ID/IdentificationDataConverter.h>
+#include <OpenMS/FORMAT/IdXMLFile.h>
+#include <OpenMS/FORMAT/MzTabFile.h>
+#include <OpenMS/FORMAT/PepXMLFile.h>
+
+///////////////////////////
+
+START_TEST(IdentificationDataConverter, "$Id$")
+
+/////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////
+
+using namespace OpenMS;
+using namespace std;
+
+START_SECTION((void importIDs(IdentificationData&, const vector<ProteinIdentification>&, const vector<PeptideIdentification>&)))
+{
+  vector<ProteinIdentification> proteins_in;
+  vector<PeptideIdentification> peptides_in;
+  IdXMLFile().load(OPENMS_GET_TEST_DATA_PATH("IdXMLFile_whole.idXML"), proteins_in, peptides_in);
+  // IdentificationData doesn't allow score types with the same name, but different orientations:
+  peptides_in[0].setHigherScoreBetter(true);
+
+  IdentificationData ids;
+  IdentificationDataConverter::importIDs(ids, proteins_in, peptides_in);
+
+  vector<ProteinIdentification> proteins_out;
+  vector<PeptideIdentification> peptides_out;
+  IdentificationDataConverter::exportIDs(ids, proteins_out, peptides_out);
+
+  TEST_EQUAL(peptides_in.size(), peptides_out.size());
+  TEST_EQUAL(proteins_in.size(), proteins_out.size());
```

Comment:
can you please also test that they are equal, e.g. `TEST_EQUAL(proteins_in == proteins_out, true)` ?

Reply:
I've added these tests, but they don't work out of the box. @hendrikweisser can you take a look as to why?

Reply:
The conversion isn't perfect and the order isn't necessarily preserved.

Reply:
@hendrikweisser can you sort and check that the main attributes are preserved and no data is lost?